pymc3/stats.py

```diff
@@ -405,16 +423,43 @@ def add_warns(*args):
 
     ics.sort(key=lambda x: x[1][0])
 
-    min_ic = ics[0][1][0]
-    Z = np.sum([np.exp(-0.5 * (x[1][0] - min_ic)) for x in ics])
+    if bootstrapping:
+        N = len(ics[0][1][3])
+
+        ic_i = np.zeros((len(ics), N))
+        for i in range(len(ics)):
+            ic_i[i] = ics[i][1][3] * N
+
+        b_weighting = dirichlet.rvs(alpha=[alpha] * N, size=b_samples,
+                                    random_state=seed)
+        weights = np.zeros((b_samples, len(ics)))
+        z_bs = np.zeros((b_samples, len(ics)))
+        for i in range(b_samples):
+            z_b = np.dot(ic_i, b_weighting[i])
+            u_weights = np.exp(-0.5 * (z_b - np.min(z_b)))
+            z_bs[i] = z_b
+            weights[i] = u_weights / np.sum(u_weights)
```

Comment:
This is pretty much `softmax`, right? That doesn't seem to be particularly stable that way, something like [this](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/extmath.py#L584) might be better.